cpp/CPPCraftDemo/CPPCraftDemo.cpp

```diff
@@ -0,0 +1,68 @@
+#include "stdafx.h"
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <assert.h>
+#include <ctime>
+#include <iostream>
+
+/**
+    Represents a Record Object
+*/
+struct QBRecord
+{
+    int recordId;
+    std::string stringValue;
+};
+
+/**
+Represents a Record Collections
+*/
+typedef std::vector<QBRecord> QBRecordCollection;
+
+/**
+    Return records that contains a string in the StringValue field
+    records - the initial set of records to filter
+    matchString - the string to search for
+*/
+QBRecordCollection QBFindMatchingRecords(QBRecordCollection records, std::string matchString)
+    {
+    QBRecordCollection result;
+    std::copy_if(records.begin(), records.end(), std::back_inserter(result), [matchString](QBRecord rec){return rec.stringValue.find(matchString) != std::string::npos; });
+    return result;
+    }
+
+/**
+    Utility to populate a record collection
+    prefix - prefix for the string value for every record
+    numRecords - number of records to populate in the collection
+*/
+QBRecordCollection populateDummyData(const std::string& prefix, int numRecords)
+    {
+    QBRecordCollection data;
+    data.reserve(numRecords);
+    for (int i = 0; i < numRecords; i++)
+        {
+        QBRecord rec = { i, prefix + std::to_string(i) };
+        data.emplace_back(rec);
+        }
+    return data;
+    }
+
+
+
+int main(int argc, _TCHAR* argv[])
+{
+    // populate a bunch of data
+    auto data = populateDummyData("testdata", 1000);
+
+    // Find a record that contains and measure the perf
+    std::clock_t startTimer = std::clock();
```

Comment:
we should use `std::steady_clock::now();`, std::clock counts the amount of time spent in all threads.